lib/spack/spack/schema/ci.py

```diff
@@ -182,18 +149,14 @@ def job_schema(name: str):
 
 
 def update(data):
-    import llnl.util.tty as tty
-
-    import spack.ci
-    import spack.environment as ev
+    """Update the data in place to remove deprecated properties.
 
-    # Warn if deprecated section is still in the environment
-    ci_env = ev.active_environment()
-    if ci_env:
-        env_config = ci_env.manifest[ev.TOP_LEVEL_KEY]
-        if "gitlab-ci" in env_config:
-            tty.die("Error: `gitlab-ci` section detected with `ci`, these are not compatible")
+    Args:
+        data (dict): dictionary to be updated
 
-    # Detect if the ci section is using the new pipeline-gen
-    # If it is, assume it has already been converted
-    return spack.ci.translate_deprecated_config(data)
+    Returns:
+        True if data was changed, False otherwise
+    """
+    # There are not currently any deprecated attributes in this section
+    # that have not been removed
+    return False
```

Comment:
You should be able to remove the entire function here. See:

https://github.com/spack/spack/blob/9ba7af404a14e84eec8f79567a6232c06a3c8d69/lib/spack/spack/config.py#L1533-L1547